consul/install.sls

```diff
@@ -0,0 +1,48 @@
+{% from "consul/map.jinja" import consul, consul_service with context %}
+
+# At the time of writing, Hashicorp doesn't maintain any official package
+# repositories for common linux distributions. The recommended installation
+# method is to download a zip, verify it, and extract consul into the system
+# path.
+
+download_package:
+  file.managed:
+    - name: /tmp/consul.zip
+    - source: {{ consul.consul_zip }}
+    - source_hash: {{ consul.consul_zip_checksum }}
+
+unzip_package:
+  module.run:
+    - name: archive.unzip
+    - zip_file: /tmp/consul.zip
+    - dest: /tmp/consul
+    - require:
+      - file: download_package
+
+install_consul_bin:
+  file.copy:
+    - name: /usr/bin/consul
+    - source: /tmp/consul/consul
+    - require:
+      - module: unzip_package
```

Comment:
The `module.run` and `file.copy` states are unnecessary if using `archive.extracted`. See above comment.